src/components/tabs/tabs.js

```diff
@@ -0,0 +1,282 @@
+import '../../vendor/polyfills/Function/prototype/bind'
+import '../../vendor/polyfills/Element/prototype/classList'
+import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
+import { nodeListForEach } from '../../common'
+
+function Tabs ($module) {
+  this.$module = $module
+  this.$tabs = $module.querySelectorAll('.govuk-tabs__tab')
+
+  this.keys = { left: 37, right: 39, up: 38, down: 40 }
+  this.jsHiddenClass = 'js-hidden'
+}
+
+Tabs.prototype.init = function () {
+  if (typeof window.matchMedia === 'function') {
+    this.setupResponsiveChecks()
+  } else {
+    this.setup()
+  }
+}
+
+Tabs.prototype.setupResponsiveChecks = function () {
+  this.mql = window.matchMedia('(min-width: 40.0625em)')
+  this.mql.addListener(this.checkMode.bind(this))
+  this.checkMode(this.mql)
+}
+
+Tabs.prototype.checkMode = function (mql) {
+  if (this.mql.matches) {
+    this.setup()
+  } else {
+    this.teardown()
+  }
+}
+
+Tabs.prototype.setup = function () {
+  var $module = this.$module
+  var $tabs = this.$tabs
+  var $tabList = $module.querySelector('.govuk-tabs__list')
+  var $tabListItems = $module.querySelectorAll('.govuk-tabs__list-item')
+
+  if (!$tabs || !$tabList || !$tabListItems) {
+    return
+  }
+
+  $tabList.setAttribute('role', 'tablist')
+
+  nodeListForEach($tabListItems, function ($item) {
+    $item.setAttribute('role', 'presentation')
+  })
+
+  nodeListForEach($tabs, function ($tab) {
+    // Set HTML attributes
+    this.setAttributes($tab)
+
+    // Handle events
+    $tab.boundTabClick = this.onTabClick.bind(this)
+    $tab.boundTabKeydown = this.onTabKeydown.bind(this)
+    $tab.addEventListener('click', $tab.boundTabClick, true)
+    $tab.addEventListener('keydown', $tab.boundTabKeydown, true)
+
+    // Remove old active panels
+    this.unhighlightTab($tab)
+    this.hidePanel($tab)
+  }.bind(this))
+
+  // If there's a tab that matches the hash
+  var $activeTab = this.getTab(window.location.hash)
+  if ($activeTab) {
```

Comment:
To avoid duplication I think this could be:

```js
var $activeTab = this.getTab(window.location.hash)
if (!$activeTab) {
  $activeTab = this.$tabs[0]
}
this.showTab($activeTab)
```

or even

```js
// Show either the active tab according to the URL's hash or the first tab
var $activeTab = this.getTab(window.location.hash) || this.$tabs[0]
this.showTab($activeTab)
```